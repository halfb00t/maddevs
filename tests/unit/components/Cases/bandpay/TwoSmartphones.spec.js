import { render, screen } from '@testing-library/vue'
import TwoSmartphones from '@/components/Cases/bandpay/TwoSmartphones'
import '../../../__mocks__/intersectionObserverMock'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const directives = {
  'lazy-load': () => {},
}

describe('BandPay TwoSmartphones component', () => {
  it('should render correctly', () => {
    const { container } = render(TwoSmartphones, {
      mocks,
      directives,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(TwoSmartphones, {
      mocks,
      directives,
    })

    expect(screen.getByText(/To deposit funds and start a project/i).className).toBeTruthy()
  })

  it('should render image', () => {
    render(TwoSmartphones, {
      mocks,
      directives,
    })

    const image = screen.getByTestId('test-picture-img')
    expect(image.getAttribute('data-src')).toBe(mocks.$getMediaFromS3())
    expect(image.getAttribute('alt')).toBe('The BandPay Mobile Applications.')
  })
})
