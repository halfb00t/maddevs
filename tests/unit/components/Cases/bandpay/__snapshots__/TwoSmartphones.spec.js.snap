// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandPay TwoSmartphones component should render correctly 1`] = `
<div>
  <section
    class="container_full background-color-silver p-48_top p-48_bottom m-96_bottom media-m-48_bottom"
  >
    <div
      class="container_regular"
    >
      <p
        class="case_paragraph m-48_bottom two-smartphones__paragraph"
      >
        
      To deposit funds and start a project, a buyer should open the app and enter the card details.
      The funds will be deposited to the sellerâ€™s BandPay account. To receive the payment,
      the seller should add a valid US non-prepaid debit card. Once the milestone is completed
      and approved by the buyer, the money will be sent to the seller automatically.
      The seller will see the amount and the number of the milestone and agreement.
    
      </p>
       
      <div
        class="two-smartphones__image-wrapper"
      >
        <picture>
          <source
            class="image"
            data-srcset="img.jpg ,img.jpg"
            type="image/webp"
          />
           
          <img
            alt="The BandPay Mobile Applications."
            class="image"
            data-src="img.jpg"
            data-srcset="img.jpg"
            data-testid="test-picture-img"
            height="652"
            width="669"
          />
        </picture>
      </div>
    </div>
  </section>
</div>
`;
